service-userlayer/src/main/java/org/oskari/map/userlayer/input/MIFParser.java

```diff
@@ -40,6 +41,9 @@ public SimpleFeatureCollection parse(File file, CoordinateReferenceSystem source
             if (crs != null) {
                 sourceCRS = crs;
             }
+            if (sourceCRS == null) {
+                throw new ServiceException("Failed to parse MIF", JSONHelper.createJSONObject("cause", "no_source_crs"));
```

Comment:
Maybe move this check into `FeatureCollectionParser.read` function? It already checks that sourceCRS can not be null inside getTransform() but could also handle the invalid/unkown crs case.